ibis/pandas/aggcontext.py

```diff
@@ -252,19 +252,41 @@ def agg(self, grouped_data, function, *args, **kwargs):
         pass
 
 
-def make_applied_function(function, args=None, kwargs=None):
+def wrap_for_apply(function, args, kwargs):
+    """Wrap a function so that it may be used with Pandas `apply`"""
     assert callable(function), 'function {} is not callable'.format(function)
 
     @functools.wraps(function)
-    def apply(
-        data,
-        function=function,
-        args=args if args is not None else (),
-        kwargs=kwargs if kwargs is not None else {},
-    ):
+    def wrapped_func(data, function=function, args=args, kwargs=kwargs):
+        return function(data, *args, **kwargs)
+
+    return wrapped_func
+
+
+def wrap_for_agg(function, args, kwargs):
```

Comment:
I am still pretty confused what's the diff between `wrap_for_apply` and `wrap_for_agg`.

Reply:
`wrap_for_apply` takes a function and returns a function that can be passed to Pandas `apply` and behave how we want (nothing special here--Pandas `apply` behavior is good as-is for the purposes of our `AggregationContexts`)

`wrap_for_agg` takes a function and returns a function that can be passed to Pandas `agg` and behave how we want (in this case, we want Pandas `agg` to never behave like Pandas `apply`).

To accomplish this, `wrap_for_agg` enforces that the function is being passed a Series (or a Series-like value, not sure, will have to look into this)

Reply:
Implementation wise `wrap_for_apply` and `wrap_for_agg` looks the same, except for that `wrap_for_agg` has a type checking. So I am still not clear what's the purpose of separating the two?

Reply:
We can technically combine `wrap_for_apply` and `wrap_for_agg` into one function, the combined function would just need a flag argument to enable/disable the type enforcing (need to disable it if the resulting function is to be passed to Pandas `apply`, and enable it if the resulting function is to be passed to Pandas `agg`).

I think this way (separate functions) is clearer though

Reply:
Fine to separate. What's not clear to at first glance that `wrap_for_agg` and `wrap_for_apply` looks the same except for `wrap_for_agg` has some type checking inside. Which doesn't map directly to what you are describing in the comment:

```
In some cases, Pandas `agg` will try to behave like Pandas `apply`.
`function` will be wrapped in such a way that Pandas will be forced
not to do this (in other words, forced to treat `function` as an
aggregation function, not a mapping function).
```

This is the main point of confusion for me. It's not clear to me `whether input data is pd.Series` relates to `the function maybe used with agg/apply`

Reply:
I re-wrote the docstring more explicit about Pandas `agg`'s behavior and how we're trying to force it to treat the function as an aggregation function using the type-checking logic

Let me know how it reads now

Reply:
Much clearer what's going on.